Source/WebCore/html/HTMLDialogElement.cpp

```diff
@@ -125,6 +144,8 @@ void HTMLDialogElement::close(const String& result)
         element->focus(options);
     }
 
+    queueDialogToggleEventTask(DialogState::Open, DialogState::Closed);
+
     queueTaskToDispatchEvent(TaskSource::UserInteraction, Event::create(eventNames().closeEvent, Event::CanBubble::No, Event::IsCancelable::No));
```

Comment:
the close event should be emitted before toggle, to get consistent event ordering (otherwise toggle might fire first when no coalescing is happening, but after when it isn't)